src/platform/plugins/shared/data/public/actions/filters/create_filters_from_value_click.ts

```diff
@@ -129,6 +134,68 @@ export const createFilter = async (
   return filter;
 };
 
+export const createFilterESQL = async (
+  table: Pick<Datatable, 'rows' | 'columns'>,
+  columnIndex: number,
+  rowIndex: number
+) => {
+  if (
+    !table ||
+    !table.columns ||
+    !table.columns[columnIndex] ||
+    !table.columns[columnIndex].meta ||
+    !table.columns[columnIndex].meta.sourceParams?.sourceField
+  ) {
+    return;
+  }
+  const column = table.columns[columnIndex];
+  const { indexPattern, sourceField, operationType, interval } = table.columns[columnIndex].meta
+    .sourceParams as {
+    indexPattern: string;
+    sourceField: string;
+    operationType: string;
+    interval: number;
+  };
+
+  const value: any = rowIndex > -1 ? table.rows[rowIndex][column.id] : null;
+  if (value === null || value === undefined) {
+    return;
+  }
+
+  const filters: Filter[] = [];
+
+  if (operationType === 'date_histogram') {
+    filters.push(
+      buildSimpleNumberRangeFilter(
+        sourceField,
+        {
+          gte: moment(value).toISOString(),
+          lt: moment(value).add(interval).toISOString(),
```

Comment:
Not sure if this is helpful but here's some code with handling time zones when creating filters in KQL:

https://github.com/elastic/kibana/blob/bb877cff7ef24c3f7295bdcdbdebb960c7db81b3/src/platform/packages/shared/kbn-es-query/src/kuery/functions/is.ts#L178-L180

Reply:
seems more complicated than what we need at the moment ?